lib/js/components/Buttons/Button/Button.vue

```diff
@@ -37,129 +38,77 @@
 			/>
 		</span>
 		<wnl-icon
-			v-if="state === STATES.LOADING"
+			v-if="state === BUTTON_STATES.LOADING"
 			class="ds-button__loadingIcon"
 			:icon="ICONS.FAD_SPINNER_THIRD"
 			:size="loadingIconSize"
 		/>
-	</span>
+	</component>
 </template>
 
 <!-- Looking for styles? These are defined in the global styles file: design-system.scss > _buttons.scss -->
 
-<script lang="ts">
-import { Value } from '../../../utils/type.utils';
-
-import WnlIcon, { ICON_SIZES, ICONS } from '../../Icons/Icon';
+<script lang="ts" setup>
+import WnlIcon, { ICON_SIZES, IconItem, ICONS, IconSize } from '../../Icons/Icon';
 import {
 	BUTTON_COLORS,
 	BUTTON_ELEVATIONS,
 	BUTTON_RADIUSES,
 	BUTTON_SIZES,
 	BUTTON_STATES,
 	BUTTON_TYPES,
+	ButtonColor,
+	ButtonRadius,
+	ButtonSize,
+	ButtonState,
+	ButtonType,
+	ButtonElevation,
 } from './Button.consts';
+import { computed } from 'vue';
+
+const {
+	size = BUTTON_SIZES.MEDIUM,
+	type = BUTTON_TYPES.FILLED,
+	color = BUTTON_COLORS.PRIMARY,
+	radius = BUTTON_RADIUSES.CAPSULE,
+	state = BUTTON_STATES.DEFAULT,
+	iconLeft = null,
+	iconRight = null,
+	elevation = BUTTON_ELEVATIONS.NONE,
+	as = 'span',
+} = defineProps<{
+	// Fixme - remove `string` from types - it is used only to avoid breaking changes for now
```

Comment:
did you identify such cases, where we used wrong prop value?

Reply:
not wrong value, but a value typed as string which caused TS errors